bentoml/_internal/io_descriptors/image.py

```diff
@@ -192,18 +213,47 @@ async def from_http_request(self, request: Request) -> ImageType:
         mime_type = content_type.decode().lower()
         if mime_type == "multipart/form-data":
             form = await request.form()
-            bytes_ = await next(iter(form.values())).read()
-        elif mime_type.startswith("image/") or mime_type == self._mime_type:
+
+            found_mimes = []
+
+            for val in form.values():
+                val_content_type = val.content_type  # type: ignore (bad starlette types)
+                if isinstance(val, UploadFile):
+                    found_mimes.append(val_content_type)
+
+                if val_content_type in self._accepted_mime_types or (
+                    val_content_type.lower().startswith("image/")
+                    and self._accept_all_images
+                ):
+                    bytes_ = val.read()
+                    break
+            else:
+                if len(found_mimes) == 0:
+                    raise BadInput("no image file found in multipart form")
+                else:
+                    raise BadInput(
+                        f"no multipart image file (with mime type in {self._accepted_mime_types}"
+                        + " or that matches 'image/*'"
+                        if self._accept_all_images
+                        else ""
+                        + "), got files with content types {', '.join(found_mimes)}"
```

Comment:
This ended up quite ugly and I wouldn't be against just factoring this out into its own if statement, I think.

Reply:
+1 for refactoring this out.